pkg/agent/monitortool/monitor.go

```diff
@@ -0,0 +1,493 @@
+	// latencyStore is the cache to store the latency of each nodes.
+	latencyStore *LatencyStore
+	// latencyConfig is the config for the latency monitor.
+	latencyConfig *LatencyConfig
+	// latencyConfigChanged is the channel to notify the latency config changed.
+	latencyConfigChanged chan struct{}
+	// gatewayConfig is the traffic encap mode for the latency monitor,
+	// indicate if the Antrea Agent is running in network policy only mode.
+	trafficEncapMode config.TrafficEncapModeType
+	// isIPv4Enabled is the flag to indicate if the IPv4 is enabled.
+	isIPv4Enabled bool
+	// isIPv6Enabled is the flag to indicate if the IPv6 is enabled.
+	isIPv6Enabled bool
+
+	// The map of node name to node info, it will changed by node watcher
+	nodeInformer coreinformers.NodeInformer
+	// nodeLatencyMonitorInformer is the informer for the NodeLatencyMonitor CRD.
+	nodeLatencyMonitorInformer crdinformers.NodeLatencyMonitorInformer
+}
+
+// LatencyConfig is the config for the latency monitor.
+type LatencyConfig struct {
+	// Enable is the flag to enable the latency monitor.
+	Enable bool
+	// Interval is the interval time to ping all nodes.
+	Interval time.Duration
+}
+
+func NewNodeLatencyMonitor(nodeInformer coreinformers.NodeInformer,
+	nlmInformer crdinformers.NodeLatencyMonitorInformer,
+	nodeConfig *config.NodeConfig,
+	trafficEncapMode config.TrafficEncapModeType) *NodeLatencyMonitor {
+	m := &NodeLatencyMonitor{
+		nodeConfig:                 nodeConfig,
+		trafficEncapMode:           trafficEncapMode,
+		latencyStore:               NewLatencyStore(trafficEncapMode.IsNetworkPolicyOnly()),
+		latencyConfig:              &LatencyConfig{Enable: false},
+		latencyConfigChanged:       make(chan struct{}, 1),
+		nodeInformer:               nodeInformer,
+		nodeLatencyMonitorInformer: nlmInformer,
+	}
+
+	// Get the IPv4/IPv6 enabled status
+	isIPv4Enabled, err := config.IsIPv4Enabled(m.nodeConfig, m.trafficEncapMode)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get IPv4 enabled status")
+	}
+	isIPv6Enabled, err := config.IsIPv6Enabled(m.nodeConfig, m.trafficEncapMode)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get IPv6 enabled status")
+	}
+	m.isIPv4Enabled = isIPv4Enabled
+	m.isIPv6Enabled = isIPv6Enabled
+
+	// Add node informer event handler for Node
+	nodeInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    m.onNodeAdd,
+		UpdateFunc: m.onNodeUpdate,
+		DeleteFunc: m.onNodeDelete,
+	})
+
+	// Add crd informer event handler for NodeLatencyMonitor
+	nlmInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    m.onNodeLatencyMonitorAdd,
+		UpdateFunc: m.onNodeLatencyMonitorUpdate,
+		DeleteFunc: m.onNodeLatencyMonitorDelete,
+	})
+
+	return m
+}
+
+// onNodeAdd is the event handler for adding Node.
+func (m *NodeLatencyMonitor) onNodeAdd(obj interface{}) {
+	node := obj.(*corev1.Node)
+	m.latencyStore.addNode(node)
+
+	klog.InfoS("Node added", "Node", klog.KObj(node))
```

Comment:
this is way too verbose. Imagine a cluster with 100+ Nodes.
Verbosity should be `V(4)` for this at most

Reply:
Ok, I will change the log level for debug.